be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -250,6 +250,10 @@ Status SegmentIterator::init(const StorageReadOptions& opts) {
     if (_char_type_idx.empty() && _char_type_idx_no_0.empty()) {
         _vec_init_char_column_id();
     }
+
+    if (opts.output_columns != nullptr) {
+        _output_columns = *(opts.output_columns);
```

Comment:
Where is opts.output_columns from?

Reply:
`_maybe_read_column_ids` in `NewOlapScanNode` -> `output_columns` in `ReaderParams` -> `output_columns` in `RowsetReaderContext` -> output_columns in StorageReadOptions